pkg/ingester/ingester_ring.go

```diff
@@ -122,6 +129,7 @@ func (cfg *RingConfig) RegisterFlags(f *flag.FlagSet, logger log.Logger) {
 
 	// TokenGenerator
 	f.StringVar(&cfg.TokenGenerationStrategy, prefix+tokenGenerationStrategyFlag, randomTokenGeneration, fmt.Sprintf("Specifies the strategy used for generating tokens for ingesters. Supported values are: %s.", strings.Join([]string{randomTokenGeneration, spreadMinimizingTokenGeneration}, ",")))
+	f.BoolVar(&cfg.TokenGenerationCanJoinEnabled, prefix+tokenGenerationCanJoinEnabledFlag, false, "True if ingesters should perform the token generator's CanJoin check before joining the ring. This check is token generator's implementation-specific.")
```

Comment:
I have a feedback about the naming. "Can join" is just the name of an internal function. The final user has no idea what it is. We need to name and describe this feature in a way that's easy to understand. In practice, what we do is registering the ingesters to the ring in order.

I propose to make this config option "spread-minimizing" specific:
- CLI flag / YAML field: `spread-minimizing-join-ring-in-order`
- Description: "True to allow this ingester registering tokens in the ring only after all previous ingesters (with ID lower than the current one) have already been registered. This configuration option is supported only when the token generation strategy is set to %q" (where `%q` is spreadMinimizingTokenGeneration)

I would also rename the `TokenGenerationCanJoinEnabled`  config field accordingly.

Reply:
Nice idea, thank you.